decentraland/components/presentational/AccountCreationProgress.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { StyleSheet, View, Text } from "react-native";
+import PropTypes from "prop-types";
+import AccountCreationStatus from "@constants/AccountCreationStatus";
+import LinkToEtherscan from "./LinkToEtherscan";
+const {
+  NOT_STARTED,
+  GENERATING_ACCOUNT,
+  FUNDING_WITH_ETH,
+  FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE,
+  FUNDING_WITH_MANA,
+  APPROVING_MARKETPLACE,
+  READY_TO_USE,
+} = AccountCreationStatus;
+
+export default function AccountCreationProgress(props) {
+  const { status, currentAction } = props;
+
+  const waitingForAccountSetup =
+    status !== NOT_STARTED && status !== READY_TO_USE;
+
+  let waitingMessage = "";
+  switch (status) {
+    case NOT_STARTED:
+      break;
+    case GENERATING_ACCOUNT:
+      waitingMessage = "Generating account...";
+      break;
+    case FUNDING_WITH_ETH:
+      waitingMessage = "Funding account with ETH...";
+      break;
+    case FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE:
+      waitingMessage = "Funding account with MANA and approving marketplace...";
+      break;
+    case FUNDING_WITH_MANA:
+      waitingMessage = "Funding account with MANA...";
+      break;
+    case APPROVING_MARKETPLACE:
+      waitingMessage = "Approving marketplace...";
+      break;
+    case READY_TO_USE:
+      break;
+    default:
+  }
+
+  return (
+    <React.Fragment>
+      {waitingForAccountSetup ? (
+        <View style={styles.textRow}>
+          <Text>{waitingMessage}</Text>
+          <LinkToEtherscan action={currentAction} />
```

Comment:
Presumably we'll be linking to the tx hash of the previous transaction that completed, right? Or are we linking them to a pending tx that has been broadcast? I ask because if it's for previous transactions, we might need to change the messages above to be about the prior tx.

Reply:
It is how it's been doing now:

```javascript
      const fundWithEthers = async accountAddress => {
        const action = fundAccountWithEthers(accountAddress);
        setAccountCreationCurrentAction(action);
        await action.waitForNonceToUpdate();
        showInfo(`Account funded with ETH`);
        setAccountCreationStatus(FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE);
      };
```
I've tried to link to the transaction that the user is waiting for.

Reply:
Okay, that makes sense! Sounds good! Resolving this.